### PR TITLE
Add details on Consumer Context to documentation

### PR DESCRIPTION
**Pull request by awholegunch**
Additions to glossary, and additions to metrics.

Fix for https://issues.cask.co/browse/CDAP-3277

Building:
- Metrics RESTful API
- Glossary

### CODE REVIEW

File: cdap-docs/reference-manual/source/glossary.rst [hunk 1]
```diff
@@ -154,7 +154,15 @@ Glossary
...
-   .. ETL and Application Templates
+   Producer
+      A producer is an entity that emits events, such as a :term:`stream` or a
+      :term:`flowlet`. Flowlets can at the same time be both :term:`consumers <consumer>`
+      and producers.
+
+   Consumer
+      A consumer is an entity that is a recipient of events, typically a :term:`flowlet`.
+      A consumer flowlet is the recipient of a queue. The difference between the total of
+      the :term:`producers <producer>` and the consumers is the pending events.
```
Comment (gokulavasan) [inline cdap-docs/reference-manual/source/glossary.rst]: difference between the total of produced events and the consumed events is the pending events?

File: cdap-docs/reference-manual/source/http-restful-api/metrics.rst [hunk 2]
```diff
@@ -65,7 +65,15 @@ Available Contexts
 The context of a metric is typically enclosed into a hierarchy of contexts. For example,
 the flowlet context is enclosed in the flow context, which in turn is enclosed in the
 application context. A metric can always be queried (and aggregated) relative to any
-enclosing context. These are the available application contexts of CDAP:
+enclosing context. 
+
+The *consumer* context relates to entities that are a recipient of events, typically a
+flowlet. A consumer flowlet is the recipient of a queue. Conversely, *producers* are
+entities that emits events, such as a stream or a flowlet. Flowlets can at the same time
+be both consumers and producers. The difference between the total of the producers and the
```
Comment (gokulavasan) [inline cdap-docs/reference-manual/source/http-restful-api/metrics.rst]: same comment as above. its the difference between produced events and consumed events. not difference of the producers and consumers isn't it?

File: cdap-docs/reference-manual/source/glossary.rst (hunk 1 above)
Comment (awholegunch) [inline cdap-docs/reference-manual/source/glossary.rst]: Yes, you're right. My mistake.

File: cdap-docs/reference-manual/source/http-restful-api/metrics.rst (hunk 2 above)
Comment (awholegunch) [inline cdap-docs/reference-manual/source/http-restful-api/metrics.rst]: Yes, you're right. My mistake.

**Comment by awholegunch** [comment]
@gokulavasan I've fixed those two comments. Please take a look.

**Comment by gokulavasan** [comment]
LGTM :+1:

**Pull request merged**